src/server/editorServices.ts

```diff
@@ -851,21 +854,31 @@ namespace ts.server {
             return { success: true, projectOptions, configFileErrors: errors };
         }
 
-        private exceededTotalSizeLimitForNonTsFiles<T>(options: CompilerOptions, fileNames: T[], propertyReader: FilePropertyReader<T>) {
+        private exceededTotalSizeLimitForNonTsFiles<T>(name: string, options: CompilerOptions, fileNames: T[], propertyReader: FilePropertyReader<T>) {
             if (options && options.disableSizeLimit || !this.host.getFileSize) {
                 return false;
             }
+
+            let availableSpace = maxProgramSizeForNonTsFiles;
+            this.projectToSizeMap[name] = 0;
+            for (const key in this.projectToSizeMap) {
+                availableSpace -= (this.projectToSizeMap[key] || 0);
+            }
+
             let totalNonTsFileSize = 0;
             for (const f of fileNames) {
                 const fileName = propertyReader.getFileName(f);
                 if (hasTypeScriptFileExtension(fileName)) {
                     continue;
                 }
                 totalNonTsFileSize += this.host.getFileSize(fileName);
-                if (totalNonTsFileSize > maxProgramSizeForNonTsFiles) {
+                if (totalNonTsFileSize > availableSpace) {
+                    this.projectToSizeMap[name] = totalNonTsFileSize;

```

Comment:
Also, does the error message indicate which projects aren't enabled? Or does it just give you a global error that the language service is disabled? If we do enable/disable on a project-by-project basis, then the error is misleading.